### PR TITLE
Convert Berendsen to v3 API

### LINKED ISSUE #864: Port Berendsen to the v3 API
## Description

Port the Berendsen integrator to the v3 API and add tests.

### PR DESCRIPTION
**Pull request by danevans09**
## Description

This converts the `Berendsen` method to the v3 API. I added an `_attach `method and added the parameter dict properties to the pybind export of the C++ class. I removed the `randomize_velocities` method, as there is no function related to randomizing velocities in _hoomd/md/TwoStepBerendsen.cc_.
## Motivation and context

The Berendsen method has not yet been converted to the v3 API

Resolves #864 

## How has this been tested?

I added the same tests as the other method classes for testing the parameters before and after attaching.

## Change log

```

```

## Checklist:

- [X] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [X] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [X] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/md/methods.py [hunk 1]
```diff
@@ -913,15 +913,19 @@ def _attach(self):
         super()._attach()
 
 
-class berendsen(_Method):
+class Berendsen(_Method):
     R""" Applies the Berendsen thermostat.
```
Comment (b-butler) [inline hoomd/md/methods.py:918]: Should we warn that this does not sample the canonical distribution properly, @joaander?

Comment (vyasr) [reply hoomd/md/methods.py:918]: I think HOOMD generally assumes that users choosing a thermostat know what they're doing wrt to the physics at this level, right? Or are we trying to provide guidance on what thermostat/barostat/etc to use v3? That seems out of scope for library documentation, but I might have missed that decision.

Comment (joaander) [reply hoomd/md/methods.py:918]: Agreed. The library documentation should state what it is and what the parameters are.

File: hoomd/md/pytest/test_methods.py [hunk 2]
```diff
@@ -68,6 +68,57 @@ def test_brownian_attributes_attached(simulation_factory,
     assert brownian.alpha == 0.125
 
 
+def test_berendsen_attributes():
```
Comment (joaander) [inline hoomd/md/pytest/test_methods.py:71]:
```suggestion
@pytest.mark.serial
def test_berendsen_attributes():
```
Berendsen fails in MPI.

File: hoomd/md/pytest/test_methods.py [hunk 3]
```diff
@@ -68,6 +68,57 @@ def test_brownian_attributes_attached(simulation_factory,
...
+
+    ramp = hoomd.variant.Ramp(1, 2, 1000000, 2000000)
+    berendsen.kT = ramp
+    assert berendsen.kT is ramp
+
+    berendsen.tau = 1.2
+    assert berendsen.tau == 1.2
+
+
+def test_berendsen_attributes_attached(simulation_factory,
```
Comment (joaander) [inline hoomd/md/pytest/test_methods.py:93]:
```suggestion
@pytest.mark.serial
def test_berendsen_attributes_attached(simulation_factory,
```

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks!
I've made some change requests below.

**Review by joaander — APPROVED** [review]
Thanks!

**Review by b-butler — APPROVED** [review]
LTGM

**Pull request merged**